test/dsets.c

```diff
@@ -5681,6 +5681,308 @@ test_multiopen(hid_t file)
+            goto error;
+
+        PASSED();
+
+        /*----------------------------------------------------------------------
+         * STEP 1: Test writing to it.
+         *----------------------------------------------------------------------
+         */
+        TESTING("    double (write)");
+
+        if (H5Dwrite(dataset, H5T_NATIVE_DOUBLE, H5S_ALL, H5S_ALL, H5P_DEFAULT, orig_data) < 0)
+            goto error;
+        PASSED();
+
+        /*----------------------------------------------------------------------
+         * STEP 2: Try to read the data we just wrote.
+         *----------------------------------------------------------------------
+         */
+        TESTING("    double (read)");
+
+        /* Read the dataset back */
+        if (H5Dread(dataset, H5T_NATIVE_DOUBLE, H5S_ALL, H5S_ALL, H5P_DEFAULT, new_data) < 0)
+            goto error;
+
+        /* Check that the values read are the same as the values written
+         * Assume size of long long = size of double
+         */
+        for (i = 0; i < (size_t)size[0]; i++) {
+            for (j = 0; j < (size_t)size[1]; j++) {
+                if (isnan(orig_data[i][j]))
+                    continue; /* skip if value is NaN */
+                if (!H5_DBL_ABS_EQUAL(new_data[i][j], orig_data[i][j])) {
+                    H5_FAILED();
+                    printf("    Read different values than written.\n");
+                    printf("    At index %lu,%lu\n", (unsigned long)i, (unsigned long)j);
+                    goto error;
+                }
+            }
+        }
+
+        PASSED();
+
+        /*----------------------------------------------------------------------
+         * Cleanup
+         *----------------------------------------------------------------------
+         */
+        if (H5Tclose(datatype) < 0)
+            goto error;
+        if (H5Sclose(space) < 0)
+            goto error;
+        if (H5Dclose(dataset) < 0)
+            goto error;
+    }
+#if H5_SIZEOF_LONG_DOUBLE != H5_SIZEOF_DOUBLE
+    /* long double */
+    {
+        long double orig_data[2][5] = {
+            {(long double)1.6081706885101836e+600L, (long double)-255.3209917099448032099170994480,
+             (long double)1.2677579992621376e-610L, (long double)64568.289448797700289448797700,
+             (long double)-1.0619721778839084e-750L},
+            {(long double)2.1499497833454840991499497833454840e+560L,
+             (long double)6.6562295504670740996562295504670740e-3,
+             (long double)-1.5747263393432150995747263393432150,
+             (long double)1.0711093225222612990711093225222612,
+             (long double)-9.8971679387636870998971679387636870e-1}};
+        long double new_data[2][5];
+
+        TESTING("    long double (setup)");
+
+        /* Define user-defined quad-precision floating-point type for dataset */
+        datatype  = H5Tcopy(H5T_NATIVE_LDOUBLE);
+        precision = 128;
```

Comment:
The precision is actually 123 bits here. I'm surprised setting it to 128 doesn't cause conversion accuracy issues, but we should still set it to 123 bits here so we aren't lying.

Reply:
I could try, but I don't remember if I had problems if it didn't match the sign pos. Maybe because of the offset?